### DISCUSSION TITLE
Error in loading auto keras pre-trained model

### DISCUSSION THREAD

**Discussion by krishdb38** [keras-team/autokeras#1662] [Q&A]
Hello,
auto-Keras prediction is very good in my data sets.
I trained (tuned) my models for several days  and save the best model using 

### save the best performing model to file
model.save(f'model_auto_keras{max_trials}.h5')

the file is saved without any error.

when I am trying to load the saved file. using 
model = tf.keras.models.load_model(model_path)
I am unable to load the model saying Value Error unknown layer; Custom > MultiCategoryEncoding .........
![image](https://user-images.githubusercontent.com/45994360/149437739-916b6b14-2e9c-49c6-b28f-d1c9f2dcec7b.png)

I searched a lot but can't find the MultiCategoryEncoding layer.
can anyone please help/share links with me to solve this problem?

**Comment by daraghimo** [comment]
just follow the tutorial [here](https://autokeras.com/tutorial/export/)

```
import tensorflow as tf
from tensorflow.keras.models import load_model
import autokeras as ak

```
to save:
`model.save("model_autokeras.h5")`

to load:
`loaded_model = load_model("model_autokeras", custom_objects=ak.CUSTOM_OBJECTS)`

to use after loading
`predicted_y = loaded_model.predict(tf.expand_dims(x_test, -1))`

**Reply by gkourogiorgas** [reply]
I want to do something similar but without the weights. I used this:
```
from keras.models import model_from_json
model_json = model.export_model().to_json()
with open("model.json", "w") as json_file:
    json_file.write(model_json)

json_file = open('model.json', 'r')
loaded_model_json = json_file.read()
json_file.close()
loaded_model = model_from_json(loaded_model_json)

loaded_model.compile(loss='binary_crossentropy', optimizer='rmsprop', metrics=['accuracy'])

loaded_model.fit(train_, train['d'],
      epochs=100, verbose=0,
      validation_split=0.2, shuffle=False,
      class_weight=cw(train))
```
And I got this error:
`Node: 'model/multi_category_encoding/string_lookup_218/None_Lookup/LookupTableFindV2' Table not initialized. [[{{node model/multi_category_encoding/string_lookup_218/None_Lookup/LookupTableFindV2}}]] [Op:__inference_train_function_43668109]`

I think it needs the multi category encoding mapping. But I would expect it to re-run it when `.fit` is called

Any help?
